src/utilities/__tests__/valueFromAST-test.ts

```diff
@@ -221,6 +229,15 @@ describe('valueFromAST', () => {
     );
     expectValueFrom('{ requiredBool: null }', testInputObj).to.equal(undefined);
     expectValueFrom('{ bool: true }', testInputObj).to.equal(undefined);
+    expectValueFrom('{ a: "abc" }', testOneOfInputObj).to.deep.equal({
```

Comment:
Is it worth testing a oneOf input object with one key that has a value that can't be coerced e.g. `{a: 1}`?